### DISCUSSION TITLE
Nucleo STM32H743

### DISCUSSION THREAD

**Discussion by hakanlundvall** [nihalpasham/rustBoot#75] [Q&A] [answered]
There is currently support for STM32H723 in rustBoot. I have a Nucleo STM32H743 evaluation board that I want to add support for it but I'm new to both rust and stm32 mcu's. I would appreciate some pointers on where to start and what parts of the existing H723 implementation that likely needs changes.

**Comment by nihalpasham** [comment]
@imrank03 - maybe this is something you can help with

**Reply by nihalpasham** [reply]
@hakanlundvall - just a quick note in case you didn't know. We have a matrix channel for rustBoot. The link to the channel is on the README page.

**Comment by imrank03** [comment, marked as answer]
> There is currently support for STM32H723 in rustBoot. I have a Nucleo STM32H743 evaluation board that I want to add support for it but I'm new to both rust and stm32 mcu's. I would appreciate some pointers on where to start and what parts of the existing H723 implementation that likely needs changes.

Hello @hakanlundvall, here are some general pointers to get you started:

- **Clone the Repository:**
  - Begin by cloning the rustBoot repository to your local development environment.

- **Develop Flash Driver and Create Partitions:**
  - Write a flash driver for the `STM32H743`, making necessary adjustments. Additionally, create partitions as needed. Note that the existing flash driver for the `STM32H723` can serve as a reference due to the similarities.

- **Compare Datasheets:**
  - Conduct a detailed comparison between the datasheets of `STM32H723` and `STM32H743`. Identify both the similarities and differences between the two microcontrollers.

- **Review and Modify Peripheral Drivers:**
  - Examine the existing peripheral drivers for the `STM32H723` and make required modifications for the `STM32H743`. This involves updating register addresses, interrupt handling, and other hardware-specific details.

- **Consult rustBoot Documentation:**
  - Refer to the [rustBoot documentation](https://nihalpasham.github.io/rustBoot-book/usage/stm32h723.html) for guidance throughout the adaptation process. 

If you have any questions or queries during the development process, feel free to ask.

**Comment by hakanlundvall** [comment]
Thanks, I have it running now. So far I have just made changes to the H723 example until it ran on H743. I'll try to clean it up and add it as a separate board alongside the H723, when I have time.
